aspnetcore/mvc/models/validation.md

```diff
@@ -57,6 +57,18 @@ MVC supports any attribute that derives from `ValidationAttribute` for validatio
 
 There may be instances where you need more features than built-in attributes provide. For those times, you can create custom validation attributes by deriving from `ValidationAttribute` or changing your model to implement `IValidatableObject`.
 
+## Notes on the use of the Required attribute
+
+Non-nullable [value types](/dotnet/csharp/language-reference/keywords/value-types) (such as `decimal`, `int`, `float`, and `DateTime`) are inherently required and don't need the `Required` attribute. The app performs no validation checks server-side for non-nullable types that are marked `Required`.
+
+MVC model binding, which isn't concerned with validation and validation attributes, rejects a form field submission containing a missing value or whitespace for a non-nullable type. Model binding usually ignores missing data for non-nullable types, where the form field is absent from the incoming form data.
+
+Client-side validation always requires a value for a form field that corresponds to a C# property marked `Required`. `Required` can be used to control the client-side validation error message.
```

Comment:
Please link to the other sentence about client-side validation also requiring a value for fields associated with all non-nullable types. Might be enough to end with something like "This control includes errors for non-nullable types, required as described [there]."

FYI only: MVC internally creates a `Required` attribute for non-nullable properties. So, there's nothing incorrect in this paragraph; it just leaves a gap unless users read all the documentation.

Reply:
>MVC internally creates a Required attribute for non-nullable properties.

That's good info, why not add it here?

Reply:
This is more of an implementation detail of how MVC adds the client-side validation for these properties. Besides, it happens only when the property doesn't already have a `[Required]` attribute. That makes the details even less succinct.